tests/integration.rs

```diff
@@ -7150,6 +7150,85 @@ mod common_parallel {
 
         handle_child_output(r, &output);
     }
+
+    #[test]
+    fn test_ivshmem() {
```

Comment:
it would be good to add a live-migration test since it appears to me the `ivshmem` implementation supports it.

Reply:
Unfortunately the integration test only really checks the PCI details - so I don't think adding a test of the same thing after life migration will test anything other than that we got the PCI configuration right - something I hope that the others tests should cover!

Reply:
Good callout. It sounds like the new device is not really being exercised meaningfully with this integration test. If that's the case, I think this is another reason why we need a feature gate.

As for live-migration (and snapshot/restore), if meaningful integration tests are not in place, I think it is better to explicitly mark such device not supporting these features for now, such as removing `impl Migratable for IvshmemDevice {}` etc..